xks-axum/src/xks_proxy/pkcs11.rs

```diff
@@ -51,11 +56,87 @@ use crate::{settings, SETTINGS};
 
 const P11_CONTEXT_READ_FAILURE: &str = "Failed to acquire pkcs11 context";
 
+// https://stackoverflow.com/questions/69350600/how-to-reconcile-rust-mutex-and-c-caller-supplied-locking-mechanisms
+
+#[repr(C)]
+struct MutexContainer {
+    mutex: Mutex<()>,
+    guard: Cell<Option<MutexGuard<'static, ()>>>,
+}
+
+extern "C" fn create_mutex_container(ppcontainer: CK_VOID_PTR_PTR) -> CK_RV {
```

Comment:
`extern fn` already implies `unsafe` per [Rust Programming Language](https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html).  (In contrast, `extern` block is not unsafe.)

> Functions declared within extern blocks are always unsafe to call from Rust code.

[Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=73e6791d914446e1be9b835a1e05b795)

Reply:
That's the other way around: `extern fn` does _not_ imply `unsafe`. `extern { fn foo() }` does.

Reply:
Won't changing it require you to change other types like the `struct` that has function pointers as fields? That's why I suggested it might be better to do that refactoring as a separate PR (and not necessarily now), especially given that this is not something this PR introduced.

Reply:
Turns out for these 4 newly added functions, the transitive set of changes is quite small.

Reply:
Fixed in the latest revision.